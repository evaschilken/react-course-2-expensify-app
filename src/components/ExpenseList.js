import React from 'react';
import { connect } from 'react-redux';
import ExpensesListItem from './ExpensesListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>
            ) : (
                props.expenses.map((expense, i) => (
                    <ExpensesListItem key={expense.id} {...expense} />
                ))
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList);