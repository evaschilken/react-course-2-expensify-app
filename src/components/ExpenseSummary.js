import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpenseSummary = (props) => (
    <div>
        {
            props.expenses.length === 1 ? (
                `Viewing ${props.expenses.length} expense `
            ) : (
                `Viewing ${props.expenses.length} expenses `
            )
        }
        totalling {numeral(getExpensesTotal(props.expenses) / 100).format('$0,0.00')}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseSummary);