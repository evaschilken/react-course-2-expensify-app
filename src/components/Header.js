import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <div>
       <h1>Expensify App</h1>
       <NavLink activeClassName="is-active" to="/" exact={true}>Dashboard</NavLink>
       <NavLink activeClassName="is-active" to="/create">Create Expense</NavLink>
       <NavLink activeClassName="is-active" to="/help">Help</NavLink>
    </div>
);

export default Header;