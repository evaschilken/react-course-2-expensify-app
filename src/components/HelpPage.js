import React from 'react';

const HelpPage = () => (
    <div>
        Here to help
    </div>
);

export default HelpPage;