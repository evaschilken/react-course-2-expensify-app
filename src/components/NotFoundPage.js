import React from 'react';
import { Link } from 'react-router-dom';

export const NotFoundPage = () => (
    <div>
        404 - Not found. <Link to="/">Go Home</Link>
    </div>
);

export default NotFoundPage;