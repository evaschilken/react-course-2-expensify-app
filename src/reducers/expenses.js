const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ]
        case 'REMOVE_EXPENSE':
            return state.filter(({id}) => id !== action.id)
        case 'EDIT_EXPENSE':   
            console.log("action.edit", action.edit);
            return state.map((expense) => {
                if (action.id === expense.id) {
                    return {
                        ...expense,
                        ...action.edit
                    }
                } else {
                    return expense
                }
            })
            return [
                ...state,
                action.expense
            ]
        default:
            return state;
    }
};

