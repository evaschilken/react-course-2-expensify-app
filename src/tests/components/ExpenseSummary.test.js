import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={1200} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render expenses summary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expenseTotal={1240099} />);
    expect(wrapper).toMatchSnapshot();
});
