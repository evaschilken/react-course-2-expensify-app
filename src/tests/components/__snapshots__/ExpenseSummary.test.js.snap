// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary with multiple expenses 1`] = `
<div>
  Viewing 
  3
   expenses 
  totalling 
  $12,400.99
</div>
`;

exports[`should render expenses summary with one expense 1`] = `
<div>
  Viewing 
  1
   expense 
  totalling 
  $12.00
</div>
`;
